e2e/specs/visual-regression.spec.ts

```diff
@@ -0,0 +1,52 @@
+import path from 'node:path';
+import { test, expect } from '@grafana/plugin-e2e';
+import dashboardJson from '../../provisioning/dashboards/dashboard.json';
+
+const panelScreenshotOptions = {
+  maxDiff: 0.02, // Ratio from 0.00 (0%) to 1.00 (100%)
+  stylePath: path.join(__dirname, 'show-panel-only.css'),
+};
+
+const { title: dashboardTitle, uid: dashboardUid, panels = [] } = dashboardJson;
+
+const dashboardSlug = dashboardTitle.toLowerCase()
+  .replace(/[^a-zA-Z0-9]+/g, ' ')
+  .trim()
+  .replace(' ', '-');
+
+const grafanaUrl = process.env.GRAFANA_URL || 'http://localhost:3000';
+const dashboardUrl = `${grafanaUrl}/d/${dashboardUid}/${dashboardSlug}`;
+
+panels
+  .filter(({ title }) => title !== 'Unconfigured panel')
+  .forEach(({ title, id }) => {
+    test(title, async ({ page }) => {
+      await page.goto(
+        `${dashboardUrl}?orgId=1&from=now-6h&to=now&timezone=browser&viewPanel=panel-${id}&kiosk`
+      );
+
+      const panel = page.locator(
+        `[data-viz-panel-key="panel-${id}"]`
+      );
+      await expect(panel).toBeVisible();
+
+      await expect(page.getByText('Loading').first())
+        .not
+        .toBeVisible();
+
+      const panelVisualizationSvg = page.locator(
+        `[data-testid="data-testid panel content"] svg`
```

Comment:
maybe drop `svg` to keep this generic for all other panels, and also capture the legend?

```suggestion
        `[data-testid="data-testid panel content"]`
```